### ISSUE TITLE
data format for flyingthings3D

### ISSUE THREAD

**Issue by JerryLeolfl** [NVIDIA/flownet2-pytorch#220]
Hi, thanks for your great work. I have a question when retrainning the model. Flow file in Flyingthins3D is in .ptm format, but the dataloader is set for .flo. Do you have a link for .flo flow data?

**Comment by CXMANDTXW** [comment]
I have meet the same problem, and I noticed that the size of optical flow file in the [https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html#downloads](url) is 364G, but the size of optical flow file downloaded by the link in this repo is 520G.
Hope to get some help.
By the way, I can not download the original file(torrent,75GB),because the download speed is always 0.

**Comment by littlespray** [comment]
> I have meet the same problem, and I noticed that the size of optical flow file in the [https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html#downloads](url) is 364G, but the size of optical flow file downloaded by the link in this repo is 621G(full dataset size).
> Hope to get some help.
> By the way, I can not download the original file(torrent,75GB),because the download speed is always 0.

The dataset is really hard to download...But I have succeeded:
link: https://pan.baidu.com/s/14PP68B3dayf7VNmeu9f5gg 
key: njl9

**Comment by littlespray** [comment]
I really suffered from the downloading process... finally, I bought a data downloading service in taobao.com, and upload it to the Baidu cloud. Now everyone can download it for free.

**Comment by littlespray** [comment]
If anyone wants the full dataset,
link: https://pan.baidu.com/s/1Wi-ouKJGaX3I5AKnApLxPg 
key: 15y5

**Comment by CXMANDTXW** [comment]
Nice work!
However, there is a small problem. The download speed is still only 60kb / s. It takes 21 hours to download a 4G file.

**Comment by CXMANDTXW** [comment]
It's so hard to download this dataset,even try from the baidupan.
The download speed is so low, IDM(60kb/s),baidupan(8kb/s).
I had to spend a month downloading this dataset, and I chose to give up.

**Comment by littlespray** [comment]
> It's so hard to download this dataset,even try from the baidupan.
> The download speed is so low, IDM(60kb/s),baidupan(8kb/s).
> I had to spend a month downloading this dataset, and I chose to give up.

Maybe you can try their VIP...I am really frustrated with it : (, but it will have a nice speed.

**Comment by CXMANDTXW** [comment]
VIP solve every thing.Damned Baidu disk.
Now,the download speed increased to 10MB/s.
Thank you very much.

**Comment by JerryLeolfl** [comment]
> > I have meet the same problem, and I noticed that the size of optical flow file in the [https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html#downloads](url) is 364G, but the size of optical flow file downloaded by the link in this repo is 621G(full dataset size).
> > Hope to get some help.
> > By the way, I can not download the original file(torrent,75GB),because the download speed is always 0.
> 
> The dataset is really hard to download...But I have succeeded:
> link: https://pan.baidu.com/s/14PP68B3dayf7VNmeu9f5gg
> key: njl9

Hi, bro
Is this the 75-G subdataset shown in https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html?
I downloaded the 311G full file data, but folw file is saved in a .pfm format, which has 3 channels. But what the network output is a 2-channel .flo file, where we cannot caculate loss between them. I wonder if it is .flo format flow file in this 75G subset

**Comment by littlespray** [comment]
> > > I have meet the same problem, and I noticed that the size of optical flow file in the [https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html#downloads](url) is 364G, but the size of optical flow file downloaded by the link in this repo is 621G(full dataset size).
> > > Hope to get some help.
> > > By the way, I can not download the original file(torrent,75GB),because the download speed is always 0.
> > 
> > 
> > The dataset is really hard to download...But I have succeeded:
> > link: https://pan.baidu.com/s/14PP68B3dayf7VNmeu9f5gg
> > key: njl9
> 
> Hi, bro
> Is this the 75-G subdataset shown in https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html?
> I downloaded the 311G full file data, but folw file is saved in a .pfm format, which has 3 channels. But what the network output is a 2-channel .flo file, where we cannot caculate loss between them. I wonder if it is .flo format flow file in this 75G subset

Yes. Actually many papers only use the subset for training. The 75G subset is the .flo file you need.
Also, the RGB images are in
link: https://pan.baidu.com/s/1poQ-mrHK0U6cuBg5kp4gmA 
key:  sj4d

**Comment by JerryLeolfl** [comment]
> > > > I have meet the same problem, and I noticed that the size of optical flow file in the [https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html#downloads](url) is 364G, but the size of optical flow file downloaded by the link in this repo is 621G(full dataset size).
> > > > Hope to get some help.
> > > > By the way, I can not download the original file(torrent,75GB),because the download speed is always 0.
> > > 
> > > 
> > > The dataset is really hard to download...But I have succeeded:
> > > link: https://pan.baidu.com/s/14PP68B3dayf7VNmeu9f5gg
> > > key: njl9
> > 
> > 
> > Hi, bro
> > Is this the 75-G subdataset shown in https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html?
> > I downloaded the 311G full file data, but folw file is saved in a .pfm format, which has 3 channels. But what the network output is a 2-channel .flo file, where we cannot caculate loss between them. I wonder if it is .flo format flow file in this 75G subset
> 
> Yes. Actually many papers only use the subset for training. The 75G subset contains the .flo files you need.
> Also, the RGB images are in
> link: https://pan.baidu.com/s/1poQ-mrHK0U6cuBg5kp4gmA
> key: sj4d

Yes, I have used this for training. But I found that there are a lot of invalid flow files with value 'nan'. Did you meet this problem? And do you have any way to solve it? I just overlook these data by an if function in code.

**Comment by JerryLeolfl** [comment]
> > > > I have meet the same problem, and I noticed that the size of optical flow file in the [https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html#downloads](url) is 364G, but the size of optical flow file downloaded by the link in this repo is 621G(full dataset size).
> > > > Hope to get some help.
> > > > By the way, I can not download the original file(torrent,75GB),because the download speed is always 0.
> > > 
> > > 
> > > The dataset is really hard to download...But I have succeeded:
> > > link: https://pan.baidu.com/s/14PP68B3dayf7VNmeu9f5gg
> > > key: njl9
> > 
> > 
> > Hi, bro
> > Is this the 75-G subdataset shown in https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html?
> > I downloaded the 311G full file data, but folw file is saved in a .pfm format, which has 3 channels. But what the network output is a 2-channel .flo file, where we cannot caculate loss between them. I wonder if it is .flo format flow file in this 75G subset
> 
> Yes. Actually many papers only use the subset for training. The 75G subset contains the .flo files you need.
> Also, the RGB images are in
> link: https://pan.baidu.com/s/1poQ-mrHK0U6cuBg5kp4gmA
> key: sj4d

Hi, have you used the dataset for training, It seems songthing wrong with the dataloader format provided, where it treats flow file flow[1] in into_future as a flow from im[1] to im[2]. what is your way to load the dataset?

**Comment by AllenYLJiang** [comment]
> > > > I have meet the same problem, and I noticed that the size of optical flow file in the [https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html#downloads](url) is 364G, but the size of optical flow file downloaded by the link in this repo is 621G(full dataset size).
> > > > Hope to get some help.
> > > > By the way, I can not download the original file(torrent,75GB),because the download speed is always 0.
> > > 
> > > 
> > > The dataset is really hard to download...But I have succeeded:
> > > link: https://pan.baidu.com/s/14PP68B3dayf7VNmeu9f5gg
> > > key: njl9
> > 
> > 
> > Hi, bro
> > Is this the 75-G subdataset shown in https://lmb.informatik.uni-freiburg.de/resources/datasets/SceneFlowDatasets.en.html?
> > I downloaded the 311G full file data, but folw file is saved in a .pfm format, which has 3 channels. But what the network output is a 2-channel .flo file, where we cannot caculate loss between them. I wonder if it is .flo format flow file in this 75G subset
> 
> Yes. Actually many papers only use the subset for training. The 75G subset contains the .flo files you need.
> Also, the RGB images are in
> link: https://pan.baidu.com/s/1poQ-mrHK0U6cuBg5kp4gmA
> key: sj4d

Dear littlespray,

Can you please share the disparity and disparity change parts of the flyingthings3d dataset? Thank you!

**Comment by zhang-mf** [comment]
> I really suffered from the downloading process... finally, I bought a data downloading service in taobao.com, and uploaded it to the Baidu cloud. Now everyone can download it for free.

Thank you so much for sharing the dataset! It has bothered me for quite a long time. I think you made a great contribution for the community.

**Comment by plateau1** [comment]
> If anyone wants the full dataset, link: https://pan.baidu.com/s/1Wi-ouKJGaX3I5AKnApLxPg key: 15y5

For the full dataset, can you provide its RGB image (clean and final)?  Thank you very much！

**Comment by shenjinyun** [comment]
> If anyone wants the full dataset, link: https://pan.baidu.com/s/1Wi-ouKJGaX3I5AKnApLxPg key: 15y5

hello, I wonder if you have the subset disparity file, which is FlyingThings3D_subset_disparity.tar.bz2, I am suffering from downloading it, Thanks.
